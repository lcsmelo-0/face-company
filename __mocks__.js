import configureMockStore from 'redux-mock-store'

const INITIAL_STATE = {
  data: [],
  isFetching: false,
  error: false,
}

export const STORE_MOCK = configureMockStore()({
  users: INITIAL_STATE,
  userDetail: INITIAL_STATE,
  posts: { data: [], postData: {}, newPostId: 0, isFetching: false, error: false },
  isVisible: false,
  loader: { isVisible: true },
})
