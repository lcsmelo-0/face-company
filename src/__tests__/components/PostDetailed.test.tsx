import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { STORE_MOCK } from '../../../__mocks__'
import { PostDetailed } from 'app-components'
import { UserPostsList } from 'app-models'

describe('Should render PostDetailed component with correct user response', () => {
  let subject

  const mockPostData: UserPostsList = {
    userId: 1,
    id: 1,
    title: 'My post title',
    body: 'My post content',
  }

  beforeEach(() => {
    subject = render(
      <Provider store={STORE_MOCK}>
        <PostDetailed data={mockPostData} />
      </Provider>,
    )
  })

  it('is expected to match the snapshot', () => {
    expect(subject).toMatchSnapshot()
  })
})
