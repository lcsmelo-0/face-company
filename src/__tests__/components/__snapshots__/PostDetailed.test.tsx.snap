// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PostDetailed component with correct user response is expected to match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="postDetailed"
      >
        <h3
          class="typography undefined"
        >
          My post title
        </h3>
        <p
          class="typography undefined"
        >
          My post content
        </p>
        <div
          class="postDetailed__line"
        >
          <button
            class="button primary"
          >
            Edit
          </button>
          <button
            class="button danger"
          >
            Delete
          </button>
        </div>
        <p
          class="typography error__hide"
        >
          Please, Enter valid values ​​for title and body
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="postDetailed"
    >
      <h3
        class="typography undefined"
      >
        My post title
      </h3>
      <p
        class="typography undefined"
      >
        My post content
      </p>
      <div
        class="postDetailed__line"
      >
        <button
          class="button primary"
        >
          Edit
        </button>
        <button
          class="button danger"
        >
          Delete
        </button>
      </div>
      <p
        class="typography error__hide"
      >
        Please, Enter valid values ​​for title and body
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
